### ISSUE TITLE
obmc-rest is using more than 40% of RAM

### ISSUE THREAD

**Issue by shenki** [openbmc/openbmc#147]
While running a recent build of openbmc I noticed the following:

```
  912     1 root     S     113m  46%   0% python /usr/sbin/obmc-rest
```

**Comment by bradbishop** [comment]
Are you sure that's not memory?  I have a pull request to drop the number of server threads that helped some.

**Comment by shenki** [comment]
Ah yep, you're right.

That's lots of memory. Do we expect it to use that much?

**Comment by bradbishop** [comment]
Yeah.  With less threads (2 I think it was) it went down to about 17%, which still seems high.  Rocket is pretty fast - we can probably trade up for something slower that uses less memory.  The WSGI servers are just drop in replacements and there are quite a few to pick from.

**Comment by shenki** [comment]
Thanks for the explanation, I'll leave the bug open as a reminder.

**Comment by KennethWilke** [comment]
This is a good comparison of wsgi servers, don't think they measure rocket but it might help anyway: http://nichol.as/benchmark-of-python-web-servers

I recommend [uWSGI](https://uwsgi-docs.readthedocs.org/en/latest/) though, it's pretty fast and light

**Comment by bradbishop** [comment]
Hello, 

I've seen a couple of things that might interest you, just take a look here <http://bandebywe.apachepeakconcepts.com/e4fjt>

Sincerely, bradleyb

**Comment by bradbishop** [comment]
Hello, 

I've  found some fresh  and cool  stuff, don't miss it up!  More info  here <http://crolufenki.frankparry.net/e4hyxo>

Warmly, bradleyb

**Comment by hweaving** [comment]
I prototyped adding nginx + gunicorn.  Running nginx itself seems lightweight and nice to have in general as a proxy server + static content server.

However, gunicorn seemed to take a lot of flash and RAM.  I agree with uWSGI, and/or gevent.  It was suggested that a PREFERRED_PROVIDER bitbake var could choose between uWSGI and gevent.

**Comment by bradbishop** [comment]
I've added python-gevent and set it as the default for phosphor.  I've also added enough plumbing to facilitate others wanting to try alternatives with something like:

PREFERRED_PROVIDER_obmc_wsgi_host = "my-uwsgi-andor-gunicorn-andor-nginx-packagegroup"

in another layer or from local.conf.

You would just need to then implement some kind of startup script/application for the alternative server and point it at /etc/default/obmc/wsgi_app.

As of today the patchset is still under review and hasn't been mainlined yet.
